pil-analyzer/src/pil_analyzer.rs

```diff
@@ -133,6 +146,24 @@ impl PILAnalyzer {
         }
     }
 
+    /// Adds core types if they are not present in the input.
+    /// These need to be present because the type checker relies on them.
+    fn core_types_if_not_present(&self) -> Option<PILFile> {
+        (!self.known_symbols.contains_key("std::prelude::Constr")).then(|| {
+            parse(
+                None,
+                "namespace std::prelude;
+    enum Constr {
```

Comment:
This overlaps with prelude.asm. We need the type here because otherwise, if we have a function
```
let x: -> expr = || 1;
let y = || x() = x();
```
Then the compiler will determine the type of `y` to be `-> std::prelude::Constr`, with the result that we cannot re-parse the printed and type-annotated source code.